ObjectMapper/Map.swift

```diff
@@ -0,0 +1,202 @@
+}
+
+public class BaseMap: Map {
+    public let objectMapper: ObjectMapper
+    public let direction: MappingDirection
+    public let json: JSONBox
+    
+    public convenience init(objectMapper: ObjectMapper, mappingDirection: MappingDirection, json: JSON = [:]) {
+        self.init(objectMapper: objectMapper, mappingDirection: mappingDirection, jsonBox: JSONBox(json: json))
+    }
+    
+    public init(objectMapper: ObjectMapper, mappingDirection: MappingDirection, jsonBox: JSONBox) {
+        self.objectMapper = objectMapper
+        self.direction = mappingDirection
+        self.json = jsonBox
+    }
+    
+    public subscript(path: [SubscriptType]) -> Map {
+        return ChildMap(parentMap: self, path: path)
+    }
+    
+    public subscript(path: SubscriptType...) -> Map {
+        return self[path]
+    }
+    
+    public func value<T>() -> T? {
+        return json.unbox.object as? T
+    }
+    
+    public func object<T: Mappable>() -> T? {
+        return objectMapper.map(json.unbox)
+    }
+    
+    public func objectArray<T: Mappable>() -> [T]? {
+        return objectMapper.mapArray(json.unbox)
+    }
+    
+    public func setValue<T: AnyObject>(value: T?) {
+        json.unbox.object = value ?? NSNull()
+    }
+    
+    public func setObject<T: Mappable>(object: T?) {
+        if let unwrappedObject = object {
+            json.unbox.object = objectMapper.toJSON(unwrappedObject).object
+        } else {
+            json.unbox.object = NSNull()
+        }
+    }
+    
+    public func setObjectArray<T: Mappable>(objectArray: [T]?) {
+        if let unwrappedObjectArray = objectArray {
+            json.unbox.object = objectMapper.toJSONArray(unwrappedObjectArray).object
+        } else {
+            json.unbox.object = NSNull()
+        }
+    }
+    
+    public func assignValueTo<T>(inout field: T) {
+        if let value: T = value() {
+            field = value
+        }
+    }
+    
+    public func assignValueTo<T>(inout field: T!) {
+        if let value: T = value() {
+            field = value
+        }
+    }
+    
+    public func assignValueTo<T>(inout field: T?) {
+        if let value: T = value() {
+            field = value
+        }
+    }
+    
+    public func assignObjectTo<T: Mappable>(inout field: T) {
+        if let object: T = object() {
+            field = object
+        }
+    }
+    
+    public func assignObjectTo<T: Mappable>(inout field: T!) {
+        if let object: T = object() {
+            field = object
+        }
+    }
+    
+    public func assignObjectTo<T: Mappable>(inout field: T?) {
+        if let object: T = object() {
+            field = object
+        }
+    }
+    
+    public func assignObjectArrayTo<T: Mappable>(inout field: [T]) {
+        if let objectArray: [T] = objectArray() {
+            field = objectArray
+        }
+    }
+}
+
+public class ChildMap: BaseMap {
+    public let parentMap: Map
+    public let path: [SubscriptType]
```

Comment:
More or less not needed now because of the workaround for the compiler issue as closures now capture these two. But I don't see why not have them? It is something you might want when extending the Map.